### PR TITLE
GFORMS-959 Added support for summary heading in add to list section

### PR DESCRIPTION
**Pull request by girishkamat**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Section.scala [hunk 1]
```diff
@@ -96,6 +96,7 @@ object Section {
     title: SmartString,
     description: SmartString,
     shortName: SmartString,
+    summaryName: Option[SmartString],
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Section.scala:99]: Shouldn't this be mandatory?

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Section.scala (hunk 1 above)
Comment (girishkamat) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Section.scala:99]: I have left it as Optional, so it works for existing form templates already saved in database.

Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Section.scala:99]: We don't need to keep backward functionality. That is why we have "Reload All" functionality in Admin tool.

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Section.scala:99]: Cool, will make the change then

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**